### PR TITLE
port package script to TS

### PR DESCRIPTION
**Pull request by shiftkey**
Related to #48

I can't be bothered submitting the upstream type declarations so I'm just going to stub them out here. We can revisit them when we go deeper into the various packaging flags we need to set.

### CODE REVIEW

File: script/globals.d.ts [hunk 1]
```diff
@@ -28,3 +28,37 @@ type LegalEagle = (
...
 ) => void
+
+// type declarations for electron-installer-redhat
+type RedhatOptions = {
+  src: string
+  dest: string
+  arch: string
+}
+
+type ElectronInstallerRedhat = (
```
Comment (j-f1) [inline script/globals.d.ts]: You can put a `declare module 'electron-installer-redhat'` block here and put an `export =` statement inside it to automatically use these types when importing `electron-installer-redhat`.

Comment (shiftkey) [reply script/globals.d.ts]: I had a bash at getting this in, but couldn't seem to get it right. Feel free to clean up after this PR gets merged (or target a PR onto this branch).

File: script/package.ts [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+  getWindowsInstallerName,
+  shouldMakeDelta,
+  getUpdatesURL,
+} from './dist-info'
+
+const distPath = getDistPath()
+const productName = getProductName()
 const outputDir = path.join(distPath, '..', 'installer')
 
+import * as electronInstaller from 'electron-winstaller'
```
Comment (joshaber) [inline script/package.ts]: Could we move this up to be with the other imports?

Comment (joshaber) [reply script/package.ts]: Also, this won't fail at runtime on non-Windows platforms will it?

Comment (shiftkey) [reply script/package.ts]: > Also, this won't fail at runtime on non-Windows platforms will it?

Dunno, I'm pretty sure you can install the module on non-Windows platforms, but the scripts are setup so they don't execute anything in the module. Will test.

Comment (shiftkey) [reply script/package.ts]: Tested on all the platforms - `yarn run package` works as intended.

**Comment by shiftkey** [comment]
:apple:

**Pull request merged**